### PR TITLE
Improve meds interaction recall and language-aware answers

### PR DESCRIPTION
**Pull request by natanayalo**
## Outcome
Surface medication-interaction guidance even when embeddings miss, keep citations clean, and ensure answer generation mirrors the user’s language.

## Demo
```bash
# Requires API stack running locally
curl -s http://localhost:8000/api/graph/run \
  -H 'content-type: application/json' \
  -d '{"user_id":"demo","query":"אקמול מתי משפיע?"}' \
  | jq '.state | {language, citations, reply: .messages[-1].content}'

curl -s http://localhost:8000/api/graph/run \
  -H 'content-type: application/json' \
  -d '{"user_id":"med-interact","query":"What are the interactions between Ibuprofen and Warfarin?"}' \
  | jq '.state | {alerts, citations}'
```
_(Expect Hebrew reply with NHS citation in the first command; the second surfaces interaction alerts with deduplicated, tracking-free citations.)_

## Acceptance checks
- [x] **Unit / integration**: `services/api/tests/unit/test_health.py`, `services/api/tests/unit/test_answer_gen.py`, `services/api/tests/integration/test_api_integration.py::test_e2e_medication_interaction_flow`
- [x] **i18n / locale**: Verified Hebrew prompts carry through `_generate_with_provider` and Hebrew deterministic onset answers.
- [x] **Observability**: BM25 handler assertions ensure combined med+query clauses hit fake ES; debug output shows normalized citations.
- [x] **Safety / disclaimers**: Answer generation still appends standard disclaimers/urgent lines and skips LLM for deterministic onset.

## Scope
**Included**
- BM25 fallback boosts per-medication combo clauses when kNN misses.
- Citation normalization/deduplication with fragment and `utm_*` stripping.
- Language-aware system prompts for LLM providers.
- Updated unit/integration coverage and roadmap entry merge.

**Excluded**
- Additional meds facts content beyond existing seeds.
- Telemetry counters for citation normalization hits/misses.
- Paraphrasing or LLM fallback for missing onset facts (deferred to PR 32/33).

## Data & provenance
No changes to persisted datasets; med facts remain in `seeds/med_facts.json`.

## Rollback / Flags
Revert this PR or deploy previous image; no new feature flags involved.

## Risks / Mitigations
- **Over-broad boosts** → Limited to user meds + primary query; tests guard interaction cases.
- **Citation stripping edge cases** → Unit tests cover non-string, custom schemes, and parsing failures.
- **Language regressions** → Hebrew integration test exercises the fallback path.

## Docs & follow-up
- Roadmap updated (`project/roadmap/pr-stack.md`).
- Follow-up: add telemetry for med fact misses (tracked separately).

### CODE REVIEW

File: services/api/app/graph/nodes/answer_gen.py [hunk 1]
```diff
@@ -469,11 +469,17 @@ def _build_reply(content: str, state: BodyState) -> str:
     return "\n\n".join(section for section in sections if section)
 
 
-def _generate_with_provider(provider: str, prompt: str) -> str | None:
+def _system_prompt(language: str) -> str:
+    if language == "he":
+        return "You produce concise, safety-first health summaries in Hebrew."
+    return "You produce concise, safety-first health summaries in English."
```
[the remark this answers is not in the document]
Comment (natanayalo) [reply services/api/app/graph/nodes/answer_gen.py:475]: Great callout—moved the localized system prompt strings into `LANG_CONFIG` so `_system_prompt` just looks them up like the other language bits.
